surface/model_openapiv3.go

```diff
@@ -439,3 +404,58 @@ func (b *OpenAPI3Builder) buildFromSchema(name string, schema *openapiv3.Schema)
 	log.Printf("Unimplemented: could not find field info for schema: %v", schema)
 	return nil
 }
+
+// buildFromOneOfAnyOfAndAllOf adds appropriate fields to the 'schemaType' given a new 'schemaOrRef'.
+func (b *OpenAPI3Builder) buildFromOneOfAnyOfAndAllOf(schemaOrRef *openapiv3.SchemaOrReference, schemaType *Type) {
+	// Related: https://github.com/googleapis/gnostic-grpc/issues/22
+	if schema := schemaOrRef.GetSchema(); schema != nil {
+		// Build a temporary type that has the required fields; add the fields to the current schema; remove the
+		// current type
+		fieldInfo := b.buildFromSchemaOrReference("ATemporaryTypeThatWillBeRemoved", schemaOrRef)
+		if t := findType(b.model.Types, "ATemporaryTypeThatWillBeRemoved"); t != nil {
+			for _, f := range t.Fields {
+				schemaType.Fields = append(schemaType.Fields, f)
+			}
+			b.removeType(t)
+		} else {
+			// schemaOrRef is some kind of primitive schema (e.g. of type string)
+			makeFieldAndAppendToType(fieldInfo, schemaType, "value")
+		}
+	} else if ref := schemaOrRef.GetReference(); ref != nil {
+		referencedSchemaName := validTypeForRef(ref.XRef)
+		// Make sure that the referenced type exists, before we add the fields to the current schema
```

Comment:
What happens if we _don't_ find it? Is that a possibility?

Reply:
If the referenced schema is in another API description it would break.

Additionally, I am wondering whether we should do the same loop for:
```
b.document.GetComponents().GetResponses().GetAdditionalProperties()
```
and:
```
b.document.GetComponents().GetRequestBodies().GetAdditionalProperties()
```

Reply:
@timburks WDYT about the above?

Reply:
I don't think we had seriously considered OneOf/AnyOf/AllOfs before this, so this is progress (even if it breaks for some cases). Here it looks like the model will be incomplete if the ref can't be resolved. Perhaps we could warn about that, but otherwise, I'm ok with addressing that situation later.

Reply:
Done